pkg/deployments/tasks/20230427-gauge-working-balance-helper/test/L2GaugeHelper.fork.ts

```diff
@@ -0,0 +1,155 @@
+    await task.run({ force: true });
+    workingBalanceHelper = await task.deployedInstance('GaugeWorkingBalanceHelper');
+  });
+
+  describe('getters (as deployed)', () => {
+    it('stores the veDelegationProxy', async () => {
+      expect(await workingBalanceHelper.getVotingEscrowDelegationProxy()).to.equal(veDelegationProxy.address);
+    });
+
+    it('stores the votingEscrow', async () => {
+      expect(await workingBalanceHelper.getVotingEscrow()).to.equal(votingEscrow.address);
+    });
+
+    it('indicates where to read supply from', async () => {
+      expect(await workingBalanceHelper.readsTotalSupplyFromVE()).to.be.false;
+    });
+  });
+
+  before('setup accounts', async () => {
+    [, veBALHolder] = await getSigners();
+
+    lpTokenHolder = await impersonate(LP_TOKEN_HOLDER, fp(100));
+  });
+
+  before('setup contracts', async () => {
+    const proxyTask = new Task('20230316-l2-ve-delegation-proxy', TaskMode.READ_ONLY, getForkedNetwork(hre));
+    veDelegationProxy = await proxyTask.deployedInstance('VotingEscrowDelegationProxy');
+    votingEscrow = await proxyTask.deployedInstance('NullVotingEscrow');
+
+    gauge = await task.instanceAt('ChildChainGauge', GAUGE);
+    lpToken = await deployedAt('IERC20', LP_TOKEN);
+  });
+
+  context('with no veBAL', () => {
+    const stakeAmount = fp(100);
+
+    before('stake in gauge', async () => {
+      await lpToken.connect(lpTokenHolder).transfer(veBALHolder.address, stakeAmount);
+      await lpToken.connect(veBALHolder).approve(gauge.address, MAX_UINT256);
+
+      await gauge.connect(veBALHolder)['deposit(uint256)'](stakeAmount);
+    });
+
+    it('projected balance should equal current', async () => {
+      const [currentWorkingBalance, projectedWorkingBalance] = await workingBalanceHelper.getWorkingBalances(
+        gauge.address,
+        veBALHolder.address
+      );
+
+      // Ensure we have equal balances (that are non-zero)
+      expect(projectedWorkingBalance).to.eq(currentWorkingBalance);
+      expect(projectedWorkingBalance).to.gt(0);
+    });
+
+    it('ratios should equal 1', async () => {
+      const [current, projected] = await workingBalanceHelper.getWorkingBalanceToSupplyRatios(
+        gauge.address,
+        veBALHolder.address
+      );
+
+      expect(projected).to.eq(current);
+      expect(projected).to.eq(FP_ONE);
+    });
+  });
+
+  describe('with veBAL', () => {
+    before('setup contracts', async () => {
+      vault = await new Task('20210418-vault', TaskMode.READ_ONLY, getForkedNetwork(hre)).deployedInstance('Vault');
+      votingEscrow = await deploy('MockL2VotingEscrow');
+
+      veDelegationProxy = await deploy('v2-liquidity-mining/VotingEscrowDelegationProxy', {
+        args: [vault.address, votingEscrow.address, ZERO_ADDRESS],
+      });
+
+      workingBalanceHelper = await deploy('v2-liquidity-mining/GaugeWorkingBalanceHelper', {
+        args: [veDelegationProxy.address, false],
+      });
+    });
+
```

Comment:
I don't get this, why are you redeploying these contracts, and then directly from the package?

Reply:
Maybe there's a better way to do it, but I'm redeploying because the regular task version uses the real VotingEscrow, while I want to use my `MockL2VotingEscrow`, so that I can directly set a veBAL balance (the "veBAL whale" step is then a one-liner).